### ISSUE TITLE
CacheInterceptor has problems handling boolean values

### ISSUE THREAD

**Issue by aaronschweig** [nestjs/nest#4997]
## Current behavior
While trying to use the `CacheInterceptor` with boolean return values inside the controller, the Interceptor always reexecutes the function when `false` is passed into the `memory`-Cache instead of return (the already cached value)

## Input Code
```ts
@Controller('role')
@UseInterceptors(CacheInterceptor)
export class RoleController {

  @Get('/can/:userId/:unitId')
  public async can(
    @Query('scopes', ParseArrayPipe) scopes: string[],
    @Param('userId') userId: string,
    @Param('unitId') unitId: string,
  ) {
    // Some DB interaction
    return false; // When returning true, the CacheInterceptor is working correctly
  }
}
```

## Expected behavior
The CacheInterceptor should also return `false`, when this is the cached value.

## Possible Solution

https://github.com/nestjs/nest/blob/07cb5a83e126911f1ce04f469aa0199e89691422/packages/common/cache/interceptors/cache.interceptor.ts#L47-L51
Instead of using this way to check if a value is present, it could be possible to explicitly check for `null` and `undefined`, so that a value of `false` could be returned.

## Environment

<pre><code>
Nest version: 7.1.3

 
For Tooling issues:
- Node version: 13  
- Platform:  Mac

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Thanks for reporting! Would you like to create a PR for this issue?

**Comment by aaronschweig** [comment]
Yes, i will create a PR for this. I have a little question regarding the tests: 

I didn't find any tests for the CacheInterceptor at all and i didn't find any reference test for an interceptor to write a new one. Do you know anyone who can provide any help there?

**Comment by kamilmysliwiec** [comment]
BTW This should be fixed in 7.3.0
